Quantum-Control-Applications/Superconducting/Pound Drever Hall Mesurement with Octave/Cross_correlation_noise_analyzer.py

```diff
@@ -0,0 +1,239 @@
+"""
+Cross_correlation_noise_analyzer.py:
+
+Version: 0.1
+
+"""
+## Imports
+from qm.QuantumMachinesManager import QuantumMachinesManager
+from qm.qua import *
+from PDH_config_octave import *
+import matplotlib.pyplot as plt
+from qm import SimulationConfig # It is also possible to simulate the IQ signals
+from qm.octave import *
+from scipy.fft import fft, fftfreq
+import os
+
+## Octave configuration
+octave_config = QmOctaveConfig()
+octave_config.add_device_info('octave1', qop_ip, octave_port)
+octave_config.set_opx_octave_mapping([("con1", "octave1")])
+octave_config.set_calibration_db(os.getcwd()) # The database can be placed anywhere you like
+qmm = QuantumMachinesManager(host=qop_ip, port=opx_port, octave=octave_config)
+
+qmm.octave_manager.set_clock("octave1", ClockType.External, ClockFrequency.MHZ_10)
+
+qm = qmm.open_qm(config)
+
+element="RR"
+qm.octave.set_lo_frequency(element, rr_LO)
+qm.octave.set_lo_source(element, OctaveLOSource.Internal)
+qm.octave.set_rf_output_gain(element, -4)
+qm.octave.set_rf_output_mode(element, RFOutputMode.on)
+
+qm.octave.set_qua_element_octave_rf_in_port(element,"octave1", 1)
+qm.octave.set_downconversion(element, lo_source=RFInputLOSource.Internal, if_mode_i=IFMode.envelope, if_mode_q=IFMode.envelope) #Dmd2LO
+
+qm.octave.calibrate_element(element, [(rr_LO, rr_IF)])
+
+qm = qmm.open_qm(config)
+
+## Cross Correlation Noise Analyzer Code
+pound_samples_per_chunk = 1*int(2*250e6/pound_modulation_freq) #in clocks = 100ns to catch 2 periods of the PDH signal
+pound_vector_size = int(pound_pulse_length/4/pound_samples_per_chunk)
+correlation_length = 4096
+averaging_length = 8
+fft_averaging_length = 10
+with program() as noise_analyzer:
+    adc_st = declare_stream(adc_trace=True)
+    adc_st2 = declare_stream(adc_trace=True)
+
+    pound_signalI = declare(fixed)
+    pound_signalQ = declare(fixed)
+    dt = declare(int)
+    pound_vector_I1 = declare(fixed, value=np.zeros(pound_vector_size))
+    pound_vector_Q1 = declare(fixed, value=np.zeros(pound_vector_size))
+    pound_vector_I2 = declare(fixed, value=np.zeros(pound_vector_size))
+    pound_vector_Q2 = declare(fixed, value=np.zeros(pound_vector_size))
+
+    pound_vector_I1_st = declare_stream()
+    pound_vector_Q1_st = declare_stream()
+    pound_vector_I2_st = declare_stream()
+    pound_vector_Q2_st = declare_stream()
+
+
+    pound_index = declare(int)
+    index = declare(int)
+    j = declare(int)
+    j2 = declare(int)
+    avg = declare(int)
+    fft_avg = declare(int)
+
+    corr_I = declare(fixed, value=np.zeros(correlation_length))
+    corr_Q = declare(fixed, value=np.zeros(correlation_length))
+    corr_I_st = declare_stream()
+    corr_Q_st = declare_stream()
+
+    align()
+
+    play('pound_pulse', 'RR')
```

Comment:
I would change "RR" to "resonator" to make it more clear (and be consistent with the rest of the tiles)